swap_meet/vendor.py

```diff
@@ -1,2 +1,71 @@
+# from unicodedata import category
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        self.inventory = inventory
+        if self.inventory is None:
+            self.inventory = []
+    def add(self, item):
+        """
+        takes in one item
+        adds item to inventory
+        returns item that was added
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False    
+
+    def get_by_category(self, category_str):
+        category_list = []
+        for elem in self.inventory:
+            if elem.category == category_str:
+                category_list.append(elem)
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item \
+            not in friend.inventory:
+            return False
+        self.inventory.remove(my_item)
+        friend.inventory.append(my_item)
+        friend.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+
+    def swap_first_item(self, friend):
+        if self.inventory == [] or friend.inventory == []:
+            return False
+        self_first_item = self.inventory.pop(0)
+        friend_first_item = friend.inventory.pop(0)
+        self.inventory.insert(0, friend_first_item)
+        friend.inventory.insert(0, self_first_item)
+        return True    
```

Comment:
The spec does not require that the items are the first in the list after the swap.  Could you use swap_item to DRY the code here?